sdk/core/corehttp/corehttp/streaming/events.py

```diff
@@ -0,0 +1,67 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+import json
+from typing import Any, Protocol
+
+from typing_extensions import runtime_checkable
+
+
+@runtime_checkable
+class EventType(Protocol):
+    """Protocol for event types."""
+
+    data: str
+    """The event data."""
+
+    def json(self) -> Any:
+        """Parse the event data as JSON.
+
+        :return: The parsed JSON data.
```

Comment:
I had to look up how `json.load` documents this (because "The parsed JSON data" is somewhat ambiguous and the return type hint isn't that helpful here :) and it says:

> Deserialize fp to a Python object using the [JSON-to-Python conversion table](https://docs.python.org/3/library/json.html#json-to-py-table).

I think what we have here is more of an `as_dict` method. 

But having that on the base `EventType` is probably wrong, because what if the event is not some form of structured data? E.g. SSE with just vanilla strings (and not json strings - which have to be quoted/escaped)

Reply:
Removed the EventTypes / assumptions of JSON being the only type of event data.